### PR TITLE
Integ tests: RAID tests and EBS default setup test

### PR DESCRIPTION
**Pull request by demartinofra**
* Implement RAID tests:
  * RAID 0
  * RAID 1
* Test default EBS configuration
* Test home correctly mounted
* Minor fix: When retrying tests on failure user_properties were added twice causing an inconsistency in the tests result count.
* Move all storage tests into a storage directory

By submitting this pull request, I confirm that my contribution is made under the terms of the Apache 2.0 license.

### CODE REVIEW

File: tests/integration-tests/tests/storage/test_raid.py [hunk 1]
```diff
@@ -0,0 +1,85 @@
...
+@pytest.mark.usefixtures("region", "os", "instance")
+def test_raid_performance_mode(scheduler, pcluster_config_reader, clusters_factory):
+    cluster_config = pcluster_config_reader()
+    cluster = clusters_factory(cluster_config)
+    remote_command_executor = RemoteCommandExecutor(cluster)
+
+    scheduler_commands = get_scheduler_commands(scheduler, remote_command_executor)
+    mount_dir = "/raid_dir"
+    _test_raid_correctly_configured(remote_command_executor, raid_type="0", volume_size=75, raid_devices=5)
+    _test_raid_correctly_mounted(remote_command_executor, mount_dir, volume_size=74)
```
[the remark this answers is not in the document]
Comment (demartinofra) [reply tests/integration-tests/tests/storage/test_raid.py]: 5 volumes of 15GB but rounded down to 74G in the df command output

File: tests/integration-tests/tests/test_raid.py [hunk 2]
```diff
@@ -0,0 +1,85 @@
...
+    mount_dir = "/raid_dir"
+    _test_raid_correctly_configured(remote_command_executor, raid_type="1", volume_size=20, raid_devices=2)
+    _test_raid_correctly_mounted(remote_command_executor, mount_dir, volume_size=20)
+    _test_raid_correctly_shared(remote_command_executor, mount_dir, scheduler_commands)
+
+
+def _test_raid_correctly_mounted(remote_command_executor, mount_dir, volume_size):
+    logging.info("Testing raid {0} is correctly mounted".format(mount_dir))
+    result = remote_command_executor.run_remote_command(
+        "df -h -t ext4 | tail -n +2 | awk '{{print $2, $6}}' | grep '{0}'".format(mount_dir)
```
Comment (enrico-usai) [inline tests/integration-tests/tests/test_raid.py]: Why are you using double braces for the awk command?

File: tests/integration-tests/tests/test_raid.py [hunk 3]
```diff
@@ -0,0 +1,85 @@
...
+    logging.info("Testing raid correctly mounted on compute nodes")
+    verify_directory_correctly_shared(remote_command_executor, mount_dir, scheduler_commands)
+
+
+def _test_raid_correctly_configured(remote_command_executor, raid_type, volume_size, raid_devices):
+    result = remote_command_executor.run_remote_command("sudo mdadm --detail /dev/md0")
+    assert_that(result.stdout).contains("Raid Level : raid{0}".format(raid_type))
+    assert_that(result.stdout).matches(r"Array Size : .*{0}\.00 GiB".format(volume_size))
+    assert_that(result.stdout).contains("Raid Devices : {0}".format(raid_devices))
+    assert_that(result.stdout).contains("Raid Devices : {0}".format(raid_devices))
```
Comment (enrico-usai) [inline tests/integration-tests/tests/test_raid.py]: You have a duplicate row, is this correct?

Comment (demartinofra) [reply tests/integration-tests/tests/test_raid.py]: adding some redundancy in test assertions :D

It was supposed to have Active Devices. Fixed, thanks!

File: tests/integration-tests/tests/test_raid.py (hunk 2 above)
Comment (demartinofra) [reply tests/integration-tests/tests/test_raid.py]: that's to escape curly braces in format command: https://docs.python.org/3/library/string.html#format-string-syntax

**Pull request merged**